sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/AadAuthorizationTests.java

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.rx;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosDatabaseForTest;
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.util.CosmosPagedFlux;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+
+public class AadAuthorizationTests extends TestSuiteBase {
+    private final static Logger log = LoggerFactory.getLogger(AadAuthorizationTests.class);
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+
+    private final static String PARTITION_KEY_PATH = "/mypk";
+    private final String databaseId = CosmosDatabaseForTest.generateId();
+
+    protected AadAuthorizationTests() {
+    }
+
+    // Cosmos public emulator only test; this test will fail if run against Azure Cosmos endpoint at this time.
+    //   We customize the Aad token to be specifically constructed for the Cosmos public emulator only; for Azure Cosmos
+    //   the token will be requested and generated from an Azure Identity service.
+    //@Test(groups = { "emulator" }, timeOut = 10 * TIMEOUT)
```

Comment:
why are we disabling the emulator test?
could you enable it for the PR? if the emulator is readly?

Reply:
There's an issue with the CI VM which prevents the AAD authenticator to initialize properly in the Gateway. I tried to reproduce this locally but unsuccessful so far (everything passes). We also validated AAD paths e2e against test federation so we should be good to go.
I will re-enable the test once a new public emulator will ship which I will expect to work around the current authenticator initialization issue.